runtime/arc.js

```diff
@@ -23,9 +23,10 @@ import {Scheduler} from './scheduler.js';
 import {registerArc} from '../devtools/shared/arc-registry.js';
 import {Id} from './id.js';
 import {ArcDebugHandler} from './debug/arc-debug-handler.js';
+import {RecipeIndex} from './recipe-index.js';
 
 export class Arc {
-  constructor({id, context, pecFactory, slotComposer, loader, storageKey, storageProviderFactory, speculative, scheduler}) {
+  constructor({id, context, pecFactory, slotComposer, loader, storageKey, storageProviderFactory, speculative, scheduler, recipeIndex}) {
```

Comment:
a recipe index doesn't seem like it should be part of an arc, I think it would make more sense to be part of the context

Reply:
I guess it's complicated, since currently the index depends on the arc... Ideally the strategies that the index runs would not depend on the arc?

Reply:
I've done the refactoring which makes RecipeIndex no longer depend on the arc, which is conceptually cleaner, especially given that we reuse the same index in arc clones.

However the context object today is a Manifest, so we would need to introduce some "composite" context object which would hold the manifest and recipe index among other things - I will defer this to a later refactoring.